### ISSUE TITLE
Mixed precision training performance issues

### ISSUE THREAD

**Issue by etasnadi** [MIC-DKFZ/nnUNet#2353]
Is it possible to throw away the mixed precision training support using a simple environment variable setting or something?

I benchmarked the `train_step` and on an `A6000` GPU, one whole training step took 0.2 sec, while gradient scaling needs 0.12 second.

If I use an older GPU like `Titan XP` that does not have native fp16 support, one step needs 1.32 seconds including 1.2 seconds for gradient scaling.

I guess that it does not worth it at all when using older GPUs and it might also possibly be  the case with never GPUs.

**Comment by etasnadi** [comment]
Seems that in the end it does not effect the performance at all on either architectures, just measured the execution time in a wrong way (when mixed precision is turned off e.g. grad_scaler is None, the computation is deferred until the `l.detach().cpu().numpy()`, while when gradient scaling is not None, the computation takes place when `grad_scaler.step()` called, and I wrongly measured the execution time just before the `l.detach.cpu().numpy()˙), so I am closing it.
